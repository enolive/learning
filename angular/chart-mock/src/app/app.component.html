<canvas baseChart
        [data]="barChartData"
        [options]="barChartOptions"
        [type]="'line'">
</canvas>
