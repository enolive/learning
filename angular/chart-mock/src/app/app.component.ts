import { Component } from '@angular/core';
import {ChartData, ChartOptions} from 'chart.js'

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent {
  barChartData: ChartData<'line'> = {
    labels: ['First', 'Second', 'Third', 'Fifth', 'Sixth'],
    datasets: [
      {
        label: 'My Data',
        data: [100, 50, 25, -14, 80],
        fill: true,
      }
    ]
  }
  barChartOptions: ChartOptions<'line'> = {
  }
}
