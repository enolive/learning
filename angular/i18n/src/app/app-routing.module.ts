import {NgModule} from '@angular/core';
import {RouterModule, Routes} from '@angular/router';
import {WelcomeComponent} from './welcome/welcome.component';

const routes: Routes = [
  {path: 'welcome', component: WelcomeComponent},
  {path: 'welcome/:name', component: WelcomeComponent},
  {path: '', redirectTo: 'welcome', pathMatch: 'full'},
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule {
}
