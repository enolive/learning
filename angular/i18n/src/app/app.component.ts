import {Component, OnInit} from '@angular/core';
import {TranslateService} from '@ngx-translate/core';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.sass']
})
export class AppComponent implements OnInit {
  private title: string;

  constructor(private translate: TranslateService) {
    translate.setDefaultLang('en');
    translate.use('en');
    translate.onLangChange.subscribe(this.changeTitleToSetLanguage.bind(this));
  }

  ngOnInit(): void {
    this.changeTitleToSetLanguage();
  }

  private changeTitleToSetLanguage() {
    this.translate
        .get('app.title')
        .subscribe(value => this.title = value);
  }

  changeLanguageTo(languageCode: string) {
    this.translate.use(languageCode);
  }
}
