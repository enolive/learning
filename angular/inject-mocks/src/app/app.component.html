<h1>Awesome App!</h1>

<output>
  {{greetings}}
</output>
