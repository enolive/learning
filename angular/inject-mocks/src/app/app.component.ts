import {Component, inject} from '@angular/core';
import { RouterOutlet } from '@angular/router';
import {GreetingsService} from './greetings.service';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrl: './app.component.css'
})
export class AppComponent {
  constructor() {
    const greetingsService = inject(GreetingsService);
    this.greetings = greetingsService.getGreetings('World')
  }

  greetings : string
}
