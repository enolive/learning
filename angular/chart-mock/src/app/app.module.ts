import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';
import { NgChartsModule } from 'ng2-charts';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    NgChartsModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
