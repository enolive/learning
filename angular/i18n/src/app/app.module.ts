import {BrowserModule} from '@angular/platform-browser';
import {NgModule} from '@angular/core';

import {AppRoutingModule} from './app-routing.module';
import {AppComponent} from './app.component';
import {TranslateLoader, TranslateModule} from '@ngx-translate/core';
import {HttpClient, HttpClientModule} from '@angular/common/http';
import {TranslateHttpLoader} from '@ngx-translate/http-loader';
import { WelcomeComponent } from './welcome/welcome.component';

const createTranslateLoader = (http: HttpClient) =>
  new TranslateHttpLoader(http, 'assets/i18n/', '.json');

@NgModule({
  declarations: [
    AppComponent,
    WelcomeComponent
  ],
  imports: [
    BrowserModule,
    AppRoutingModule,
    HttpClientModule,
    TranslateModule.forRoot({
      loader: {
        provide: TranslateLoader,
        useFactory: (createTranslateLoader),
        deps: [HttpClient]
      }
    })
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule {
}
