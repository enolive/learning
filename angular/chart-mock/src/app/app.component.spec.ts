import {AppComponent} from './app.component'
import {render} from '@testing-library/angular'
import {Directive, Input, NgModule} from '@angular/core'
import {TestBed} from '@angular/core/testing'

describe('AppComponent', () => {
  it('works', async () => {
    await render(AppComponent, {
      imports: [
        ChartTestModule,
      ],
    })

    const values = TestBed.inject(ChartValuesDump)

    expect(values).toMatchSnapshot()
  })
})

class ChartValuesDump {
  data?: any
  type?: any
  options?: any
}

@Directive({
  selector: 'canvas[baseChart]',
  exportAs: 'base-chart',
})
class BaseChartTestDirective {
  constructor(private values: ChartValuesDump) {
  }

  @Input() set type(value: any) {
    this.values.type = value
  }

  @Input() set data(value: any) {
    this.values.data = value
  }

  @Input() set options(value: any) {
    this.values.options = value
  }
}

@NgModule({
  declarations: [BaseChartTestDirective],
  providers: [ChartValuesDump],
  exports: [BaseChartTestDirective],
})
class ChartTestModule {
}
