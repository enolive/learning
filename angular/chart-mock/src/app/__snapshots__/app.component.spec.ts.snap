// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent works 1`] = `
ChartValuesDump {
  "data": Object {
    "datasets": Array [
      Object {
        "data": Array [
          100,
          50,
          25,
          -14,
          80,
        ],
        "fill": true,
        "label": "My Data",
      },
    ],
    "labels": Array [
      "First",
      "Second",
      "Third",
      "Fifth",
      "Sixth",
    ],
  },
  "options": Object {},
  "type": "line",
}
`;
